source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,797 @@
+            vReleaseNetworkBufferAndDescriptor( pxBufferDescriptor );
+            /* Not sure if a trace is required.  The stack did not want this message */
+        }
+    }
+    else
+    {
+        FreeRTOS_printf( ( "RX No Buffer Available\n" ) );
+        ENET_ReadFrame( ENET, &( ethernetifLocal->handle ), NULL, length, 0, NULL );
+        /* No buffer available to receive this message */
+        iptraceFAILED_TO_OBTAIN_NETWORK_BUFFER();
+    }
+}
+
+
+/**
+ * @brief This function is used to setup the PHY in auto-negotiation mode.
+ *
+ * @param[out] pxConfig: the configuration parameters will be written into this pointer.
+ * @param[in|out] peEMACState: the current state in the call to xNetworkInterfaceInitialise.
+ *                   The state will be updated to the next state depending on the status of
+ *                   the PHY initialization.
+ *
+ * @return kStatus_Success if the PHY was initialized; error code otherwise.
+ */
+static status_t xSetupPHY( phy_config_t * pxConfig )
+{
+    status_t xStatus;
+
+    ethernetifLocal->RxBuffDescrip = &( rxBuffDescrip_0[ 0 ] );
+    ethernetifLocal->TxBuffDescrip = &( txBuffDescrip_0[ 0 ] );
+    ethernetifLocal->RxDataBuff = &( rxDataBuff_0[ 0 ] );
+    ethernetifLocal->TxDataBuff = &( txDataBuff_0[ 0 ] );
+    ethernetifLocal->base = ( void * ) ENET_BASE;
+
+    /* Set the clock frequency. */
+    mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+    mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+    /* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+    pxConfig->autoNeg = pdTRUE;
+    pxConfig->duplex = kPHY_FullDuplex;
+    pxConfig->phyAddr = BOARD_ENET0_PHY_ADDRESS;
+    pxConfig->speed = kPHY_Speed100M;
+
+    FreeRTOS_printf( ( "Starting PHY initialization.\r\n" ) );
+
+    xStatus = PHY_Init( &phyHandle, pxConfig );
+
+    if( xStatus != kStatus_Success )
+    {
+        FreeRTOS_printf( ( "Failed to initialize the PHY..." ) );
+    }
+
+    return xStatus;
+}
+
+/**
+ * @brief This function is used wait on the auto-negotiation completion. In case auto-negotiation
+ *        fails, the function tries to use the default values (100M speed; full duplex communication
+ *        link) to get the link up.
+ *
+ * @param[out] pxConfig: the updated configuration parameters will be written into this pointer.
+ * @param[in|out] peEMACState: the current state in the call to xNetworkInterfaceInitialise.
+ *                   The state will be updated to the next state depending on the status of
+ *                   the PHY initialization.
+ *
+ * @return kStatus_Success if the PHY was initialized; error code otherwise.
+ */
+static status_t xWaitPHY( phy_config_t * pxConfig )
+{
+    status_t xStatus;
+    bool LinkUp;
+    bool autoNegotiationComplete;
+    uint8_t ucCounter = 0;
+
+    do
+    {
+        xStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
```

Comment:
As we discussed in person yesterday.. This loop should really be reversed.. 

First checking PHY_GetLinkStatus() in case auto-negotiation is disabled. 

Then you could optionally poll PHY_GetAutoNegotiationStatus() if auto-negotiation is enabled for the purpose of providing log messages to make the state of the system clear. Ultimately, the thing that drives this state machine should be only the call to PHY_GetLinkStatus().